src/realm/parser/driver.cpp

```diff
@@ -544,6 +543,18 @@ Query EqualityNode::visit(ParserDriver* drv)
             }
         }
     }
+    else if (left_type == type_Link && right->has_multiple_values()) {
+        auto link_column = dynamic_cast<const Columns<Link>*>(left.get());
```

Comment:
I am not sure if this is the right place for the fix, but the problem is that we can't defer the comparison of links passed as list argument to the query engine and a link column. 
Since the link column will be treated as `type_Link`, whereas each link is treated as type_TypeLink (`ObjLink`) ... the 2 types are not comparable, thus the link will never be found, although present. 

The same query works if instead of a list, we pass a single link argument (as per the test added)

An alternative to eagerly parse the list of arguments could be to prohibit the query itself, but I don't think it makes sense.